### ISSUE TITLE
Oct 12 - 16 - Automate Your Development with GitHub Actions

### ISSUE THREAD

**Issue by mebaumb** [microsoft/ReactorPlanning#111]
# Event Information
| Category | Details |
|-----------|---------|
| Reactor Topic | DevOps and Dev Tools |
| Monthly Topic (if applicable) | Open Source |
| Weekly Topic (if applicable) |  GitHub |
| Workshop Duration | 60 min | 
| Title | Automate Your Development with GitHub Actions  |
| Description |<p>GitHub Actions automate many development cycle tasks like and executing unit tests, running linting, and pushing deployments to Azure. In this workshop, you'll become familiar with GitHub Actions by automating some of these tasks and workflows.</p><p>Prerequisites: This workshop recommends that you have a working knowledge of key GitHub features, such as Issues and Pull Requests. To follow along, you will need accounts for Azure (https://azure.microsoft.com/free/) and GitHub (https://github.com/join).</p> |
| Learn content | https://docs.microsoft.com/learn/modules/github-actions-automate-tasks//?WT.mc_id=reactor-github-reactor |

# Workshop(s) details
| Date |  Local Start Time |  Location |  Speaker | SharePoint ID | Platform |
|------|-------------------|-----------|----------|---------------|----------|
|Oct 13| 6:30 PM | Bangalore | TBD | 11628 | TBD |
|Oct 13| 4:00 PM PST | US/Canada/Sydney | Chris Bautista (Todd Enders - Moderator) | 11629 | TBD |
|Oct 14| 7:30 PM | Sao Paulo | TBD | 11632 | TBD |
|Oct 14| 7:30 PM | Shanghai | TBD | 11633 | TBD |
|Oct 14| 5:00 PM | London, Stockholm, Tel Aviv, Abu Dhabi | TBD | 11634 | TBD |

# Event Checklist

## 10 Weeks Prior
- [x] Reactor Content Team (RCT) opens a new issue for an upcoming local workshop filling in the Event Information and Workshop details tables. RCT member who owns content (content owner) this week will be the assignee.
- [x] RCT and content owner discuss details of the event in the comments of this issue. RCT adds a _confirmed-event_ label and _local-digital-event_ tag to the issue and adds event to the __Local Digital Events__ project. Issue will appear in the __Upcoming Streams__ column
- [x] RCT adds a _ready-for-marketing-edit_ label so the RCT can review title and description

## 8 Weeks Prior
- [x] RCT does a review of title and description and removes the _ready-for-marketing-edit_ label 
- [x] RCT moves the issue to the __Ready for SharePoint__ column
- [ ] RCT informs the Reactor Marketing Team of the new event information
- [ ] RCT confirms the speaker
- [ ] RCT provides speaker with assets (e.g. slides, overlays) and the survey data for the event
- [ ] RCT moves the issue to the __Content Preparation In Progress__ column

## 5 weeks Prior
- [ ] Content owner opens a PR with the new content for the workshop on the [customer-facing Reactor repository](https://github.com/microsoft/Reactors/tree/main/online-event-resources) and adds the _content-development_ and _local-digital-event_ tag to the PR. 
- [ ] RCT reviews, approves, and merges the changes
- [ ] RCT shares content with the vendor/instructor
- [ ] RCT provides vendor/instructor with assets (e.g. slides, overlays) and the survey data for the event
- [ ] RCT ensures that instructors have received an invite to the workshops and that PM team works with instructors to get setup in the platform (e.g. Teams, Teams Live Event)

## 1 Week Prior
- [ ] Connect vendor/instructor with the PM(s) for the locations where this event has been marketed
- [ ] Check-in with vendor/instructor that they are prepared for the workshop
- [ ] RCT moves the issue to the __Ready to Stream__ column

## During and After Stream
- [ ] Instructor (and optional moderator) run the stream
- [ ] _(Optional)_RCT submits recording to be uploaded to the Reactor YouTube and to be made available on LearnTV
- [ ] RCT moves issue to the __Stream Complete__ column

**Comment by mebaumb** [comment]
@KimberlyBolton1 - not sure if you saw this one too!

These events have been created in Dynamics (need to update with final title/abstract)

**Comment by KimberlyBolton1** [comment]
@meaghanlewis here's the edit for your review:

**Automate Your Development with GitHub Actions**
GitHub Actions automate many development cycle tasks like and executing unit tests, running linting, and pushing deployments to Azure. In this workshop, you'll become familiar with GitHub Actions by automating some of these tasks and workflows. 
Prerequisites: This workshop recommends that you have a working knowledge of key GitHub features, such as Issues and Pull Requests. To follow along, you will need accounts for Azure (https://azure.microsoft.com/free/) and GitHub (https://github.com/join).

**Comment by meaghanlewis** [comment]
Thank you @KimberlyBolton1. Edit looks great.

**Comment by meaghanlewis** [comment]
Here's an overview of what this workshop will cover and suggested agenda to follow with links and resources.

## Content to teach
This workshop will take Reactor community members through one module on Microsoft Learn.
- [Use GitHub Actions to automate tasks](https://docs.microsoft.com/learn/modules/github-actions-automate-tasks//?WT.mc_id=reactor-github-reactor)

**This is the link that should be shared with attendees during the workshop:** **https://aka.ms/AutomateWorkflowGitActions-8**

Let them know we'll only be going through the [second module](https://aka.ms/AutomateDevTasksGitHub-5), but encourage them to go through the other modules in their own time. For each completed module, they get an acheivement badge in their profile.

## Workshop steps

1. [Introduction to content](https://docs.microsoft.com/en-us/learn/modules/github-actions-automate-tasks/1-introduction)
2. [Intro to GitHub Actions](https://docs.microsoft.com/en-us/learn/modules/github-actions-automate-tasks/2-github-actions-automate-development-tasks)
    - What is GitHub Actions:
      - https://github.com/features/actions
      - https://docs.github.com/en/actions
      - https://github.com/marketplace?type=actions
    - What is a GitHub Actions workflow:
      - https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions
      - https://docs.github.com/en/actions/configuring-and-managing-workflows
3. [Create a basic container by using GitHub Actions](https://docs.microsoft.com/en-us/learn/modules/github-actions-automate-tasks/3-exercise-create-container-action//?WT.mc_id=reactor-github-reactor). 
    - This will walk through a guided exercise in GitHub's Learning Lab: https://lab.github.com/. Explain what Learning Lab is. Remind them they must have a GitHub account to follow along here.
    - Complete the learning lab, then come back to complete the [knowledge check](https://docs.microsoft.com/en-us/learn/modules/github-actions-automate-tasks/4-knowledge-check) and [summary](https://docs.microsoft.com/en-us/learn/modules/github-actions-automate-tasks/5-summary)

4. Next steps
   - Encourage attendees to complete the learning path: https://aka.ms/AutomateWorkflowGitActions-8 to get comfortable with GitHub Actions
5. Closing
    - Q&A
    - Survey: https://aka.ms/Reactor/Survey
      - Note: The event code is listed in this issue description and/or your Reactor contact can share that with the community members
    - Share Resources
    - Check if the Reactor contact has other information to share

## Resources
- [GitHub Actions documentation](https://help.github.com/actions)
- [GitHub created actions](https://github.com/actions)
- [Actions on GitHub Marketplace](https://github.com/marketplace?type=actions)
- [Metadata syntax for GitHub Actions](https://help.github.com/en/actions/building-actions/metadata-syntax-for-github-actions)
- [Workflow syntax for GitHub Actions](https://help.github.com/en/actions/reference/workflow-syntax-for-github-actions)
- [Events that trigger workflows](https://help.github.com/actions/reference/events-that-trigger-workflows)
- [Introduction to Docker Containers](https://docs.microsoft.com/en-us/learn/modules/intro-to-docker-containers/)

**Comment by KimberlyBolton1** [comment]
@mebaumb abstract final and moved to Ready for SharePoint.

**Comment by mebaumb** [comment]
Title/Abstract sent to Caio for translation

US/Canada Published

All other locations updated - will publish when we have speaker details.

**Comment by mebaumb** [comment]
Bangalore, Shanghai, Sao Paulo, and London/Stockholm/Tel Aviv/Abu Dhabi all published with out speaker details (9/2)
